backend/pkg/actions/submitflag/submitflag.go

```diff
@@ -0,0 +1,175 @@
+package submitflag
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/ctfg/ctfg/backend/gen/ctfg/public/model"
+	. "github.com/ctfg/ctfg/backend/gen/ctfg/public/table"
+	"github.com/ctfg/ctfg/backend/pkg/util"
+	"github.com/go-jet/jet/v2/postgres"
+	"github.com/google/uuid"
+	_ "github.com/lib/pq"
+	"github.com/rs/zerolog/log"
+	"io"
+	"net/http"
+	"os"
+)
+
+type ActionPayload struct {
+	SessionVariables map[string]interface{} `json:"session_variables"`
+	Input            submitFlagArgs         `json:"input"`
+}
+
+type GraphQLError struct {
+	Message string `json:"message"`
+}
+
+// TODO (chthompson) oh god why
+var (
+	db *sql.DB
```

Comment:
if you're using actions, you shouldn't need a connection to the database.

Reply:
hasura suggests having actions call back into hasura from the action when you want data https://github.com/hasura/hasura-actions-examples/blob/master/data-validations/inventory-check/functions/nodejs-express/src/handlers/createCartItem.js#L3 which is not ideal if you want to eject from hasura in the future. SQL queries are written here to not incur that debt.